etcd/tls-etcd-clients.md

```diff
@@ -0,0 +1,104 @@
+# Configure CoreOS Components to Connect to etcd with TLS
+
+This document explains how to configure CoreOS components to use secure HTTPS connections to an etcd cluster. The target etcd cluster must already be using HTTPS for its own communications, as explained in the [etcd HTTP to HTTPS migration guide][etcd-live-http-https].
+
+The primary CoreOS components that use etcd are:
+
+* [flannel][flannel]
+* [fleet][fleet]
+* [locksmith][locksmith]
+
+This document assumes three CoreOS nodes are running three etcd cluster members: Call them server1, server2, and server3, with IP addresses 172.16.0.101, 172.16.0.102, and 172.16.0.103, respectively. We further assume that the [necessary Certificate Authority (CA) and client certificate/key pairs have been created][self-signed-ca].
+
+## Configure flannel to Use Secure etcd Connection
+
+The flannel systemd unit file reads the `/run/flannel/options.env` [environment][systemd-environments] file for configuration. `options.env` is created automatically by any [cloud-config][cloud-config] that contains a `flannel:` configuration section. For example, a cloud-config with the following contents:
+
+```yaml
+coreos:
+  flannel:
+    etcd_endpoints: "https://172.16.0.101:2379,https://172.16.0.102:2379,https://172.16.0.103:2379"
+    etcd_cafile: /etc/ssl/etcd/ca.pem
+    etcd_certfile: /etc/ssl/etcd/client.pem
+    etcd_keyfile: /etc/ssl/etcd/client-key.pem
+```
+
+will generate the following `/run/flannel/options.env`:
+
+```ini
+FLANNELD_ETCD_ENDPOINTS=https://172.16.0.101:2379,https://172.16.0.102:2379,https://172.16.0.103:2379
+FLANNELD_ETCD_CAFILE=/etc/ssl/etcd/ca.pem
+FLANNELD_ETCD_CERTFILE=/etc/ssl/etcd/client.pem
+FLANNELD_ETCD_KEYFILE=/etc/ssl/etcd/client-key.pem
+```
+
+## Configure fleet to Use Secure etcd Connection
+
+The fleet systemd unit file reads `/run/systemd/system/fleet.service.d/20-cloudinit.conf` for configuration. `20-cloudinit.conf` is created by any [cloud-config][cloud-config] that contains a `fleet:` configuration section. For example, a cloud-config with the following contents:
+
+```yaml
+coreos:
+  fleet:
+    etcd_servers: "https://172.16.0.101:2379,https://172.16.0.102:2379,https://172.16.0.103:2379"
+    etcd_cafile: /etc/ssl/etcd/ca.pem
+    etcd_certfile: /etc/ssl/etcd/client.pem
+    etcd_keyfile: /etc/ssl/etcd/client-key.pem
+```
+
+will generate a `/run/systemd/system/fleet.service.d/20-cloudinit.conf` drop-in with these contents:
```

Comment:
link to drop in: https://coreos.com/os/docs/latest/using-systemd-drop-in-units.html